packages/aws-cdk/lib/commands/init/init.ts

```diff
@@ -18,103 +19,126 @@ const decamelize = require('decamelize');
 
 /**
  * Config options for the CDK init command.
- * - Built-in templates: use 'type' (e.g., 'app', 'lib')
- * - Local templates: use 'templatePath'
- * - Git templates: use 'gitUrl' (optionally with 'templateName')
- * - NPM templates: use 'npmPackage' (optionally with 'templateName')
+ *
+ * Options for different template sources:
+ * - Built-in templates: use 'type' (e.g., 'app', 'lib', 'sample-app')
+ * - Local templates: use 'fromPath'
+ * - Git templates: use 'fromGit' (optionally with 'templatePath')
+ * - GitHub shorthand: use 'fromGithub' (optionally with 'templatePath')
+ * - NPM templates: use 'fromNpm' (optionally with 'templatePath')
+ *
+ * Note: Custom templates (from Git, NPM, or local path) don't require info.json.
+ * They're detected by the presence of language directories (typescript, python, etc.).
+ * Custom templates don't support aliases - only built-in templates do.
  */
 export interface CliInitOptions {
+  // Built-in template type (app, lib, sample-app)
```

Comment:
These are good comments for context, and should be in TSDoc format. See example in https://github.com/aws/aws-cdk-cli/blob/main/packages/%40aws-cdk/toolkit-lib/lib/actions/bootstrap/index.ts#L54